python/hsfs/core/feature_group_engine.py

```diff
@@ -57,7 +57,7 @@ def save(
             return None, ge_report
 
         offline_write_options = write_options
-        online_write_options = self.get_kafka_config(write_options)
+        online_write_options = write_options
```

Comment:
yeah for the moment I left it because I was afraid some downstream calls make modifications to the original dict. however, I think since we don't make a copy, the assignment to a new variable won't help in that case either.